### PR TITLE
parser: support `admin flush plan_cache`

### PR DESCRIPTION
**Pull request by Reminiscent**
### What problem does this PR solve?
Related to https://github.com/pingcap/tidb/issues/30826
parser: support 'admin flush plan_cache'

### What is changed and how it works?
parser: support 'admin flush plan_cache'

### Check List

Tests 

- [ ] Unit test
- [x] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

File: parser/ast/misc.go [hunk 1]
```diff
@@ -1864,6 +1865,14 @@ type HandleRange struct {
 	End   int64
 }
 
+type StatementScope int
+
+const (
+	SessionScope StatementScope = iota
```
Comment (tangenta) [inline parser/ast/misc.go:1871]: It is better to add a `SessionNone` and use `0` as the default value. `session/instance/global` can be meaningless for the other `AdminStmt`.

File: parser/parser_test.go [hunk 2]
```diff
@@ -809,6 +809,11 @@ func TestDMLStmt(t *testing.T) {
 		// This case would be removed once TiDB PR to remove ADMIN RELOAD STATISTICS is merged.
 		{"admin reload statistics", true, "ADMIN RELOAD STATS_EXTENDED"},
 		{"admin reload stats_extended", true, "ADMIN RELOAD STATS_EXTENDED"},
+		// Test for 'admin flush plan_cache'
+		{"admin flush instance plan_cache", true, "ADMIN FLUSH INSTANCE PLAN_CACHE"},
+		{"admin flush session plan_cache", true, "ADMIN FLUSH SESSION PLAN_CACHE"},
+		// We do not support the global level. We will check it in the later.
+		{"admin flush global plan_cache", true, "ADMIN FLUSH GLOBAL PLAN_CACHE"},
```
Comment (djshow832) [inline parser/parser_test.go:816]: You don't support it for now or forever?

Comment (Reminiscent) [reply parser/parser_test.go:816]: According to the current plan, the global level will not be supported in the future

File: parser/ast/misc.go (hunk 1 above)
Comment (Reminiscent) [reply parser/ast/misc.go]: Update PTAL.

**Review by qw4990 — APPROVED** [review]

**Review by qw4990 — APPROVED** [review]

File: parser/ast/misc.go [hunk 3]
```diff
@@ -1864,6 +1865,15 @@ type HandleRange struct {
 	End   int64
 }
 
+type StatementScope int
+
+const (
+	NoneScope StatementScope = iota
```
Comment (tangenta) [inline parser/ast/misc.go:1871]: It is better to name it `StatementScopeNone`, which is friendly to auto-complete.

**Review by qw4990 — APPROVED** [review]

**Comment by Reminiscent** [comment]
@tangenta Update. PTAL

**Comment by tangenta** [comment]
Please run `make parser` to update `parser.go`.

**Comment by Reminiscent** [comment]
> Please run `make parser` to update `parser.go`.

@tangenta Sorry for the missed commit. PTAL

**Review by tangenta — APPROVED** [review]

**Pull request merged**